src/components/app-configure/access/EditDelegate.vue

```diff
@@ -0,0 +1,152 @@
+<script lang="ts" setup>
+import { onMounted, ref } from 'vue'
+
+import VButton from '@/components/lib/VButton/VButton.vue'
+import VCard from '@/components/lib/VCard/VCard.vue'
+import VDropdown from '@/components/lib/VDropdown/VDropdown.vue'
+import VOverlay from '@/components/lib/VOverlay/VOverlay.vue'
+import VSeperator from '@/components/lib/VSeperator/VSeperator.vue'
+import VStack from '@/components/lib/VStack/VStack.vue'
+import VTextField from '@/components/lib/VTextField/VTextField.vue'
+import { useToast } from '@/components/lib/VToast'
+import { editDelegate } from '@/services/gateway.service'
+import { useAppId } from '@/use/getAppId'
+
+const emit = defineEmits(['close', 'generateKey', 'edited'])
+
+const props = defineProps({
+  delegateInfo: {
+    type: Object,
+    required: true,
+  },
+  delegateKeys: {
+    type: Array,
+    required: true,
+  },
+})
+
+const toast = useToast()
+const appId = useAppId()
+const delegateName = ref(props.delegateInfo.name)
+const selectedDelegateAddress = ref('')
+
+onMounted(() => {
+  selectedDelegateAddress.value = props.delegateInfo.address
+})
+
+async function onSubmit() {
+  if (!delegateName.value.length || !selectedDelegateAddress.value.length) {
+    toast.error('Please fill all required values')
+  }
+  try {
+    const payload = {
+      id: props.delegateInfo.id,
+      name: delegateName.value,
+      address: selectedDelegateAddress.value,
+      permissions: ['Dowload', 'Reshare'],
```

Comment:
Will this cause an error?

```suggestion
      permissions: ['Download', 'Reshare'],
```

Reply:
Fixed it with help of Saurav